ffi/wgpu.h

```diff
@@ -669,12 +696,31 @@ typedef struct {
   bool alpha_to_coverage_enabled;
 } WGPURenderPipelineDescriptor;
 
-typedef struct {
+typedef struct WGPUChainedStruct {
```

Comment:
isn't this redefining the struct? I'm surprised this works :)

Reply:
Welcome to C! You first define the struct with a name, which puts it in the "struct" namespace, then typedef it to a normal name, which goes in the regular namespace. No idea how it interacts with C++ tbh, but thats a problem for another time.

Reply:
To clarify, I mean there is a previous definition:
```cpp
typedef struct WGPUChainedStruct WGPUChainedStruct;
```
I assumed that nothing is possible to define under this name afterwards, but here you are actually redefining the typedef. Still confused a bit, eh

Reply:
Same here tbh, I guessed this was okay because of how enums are declared with an unnamespaced name then the underlying type was also typedefed. C never ceases to confuse me.